### PR TITLE
Add agent management section to docs

### PR DESCRIPTION
**Pull request by mhausenblas**
This PR addresses https://github.com/open-telemetry/opentelemetry.io/issues/2282

---

**Preview**: https://deploy-preview-2285--opentelemetry.netlify.app/docs/collector/management/

### CODE REVIEW

File: content/en/docs/collector/management.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: Management
+weight: 23
+---
+
+Please be sure to review the following documentation:
```
Comment (cartermp) [inline content/en/docs/collector/management.md:6]:
```suggestion
Please be sure to review the following documentation a prerequisite:
```

File: content/en/docs/collector/management.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+*  [Getting Started][otel-collector-getting-started] to understand how to
+   install the OpenTelemetry collector.
+*  [Configuration][otel-collector-configuration] for how to configure the 
+   OpenTelemetry collector, setting up telemetry pipelines.
+
```
Comment (cartermp) [inline content/en/docs/collector/management.md:12]:
```suggestion
This document describes how you can manage your OpenTelemetry Collector deployment at scale.
```

File: content/en/docs/collector/management.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+## Basics
+
+Telemetry collection at scale requires us to come up with a structured approach
+to manage agents. Typical agent management tasks (ordered from most common to
+least common) include:
```
Comment (cartermp) [inline content/en/docs/collector/management.md:17]:
```suggestion
Telemetry collection at scale requires a structured approach to manage agents.
Typical agent management tasks (ordered from most common to
least common) include:
```

File: content/en/docs/collector/management.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Querying the agent information and configuration. The agent information can
+include its version, operating system related information, or capabilities.
+The configuration of the agent refers to its telemetry collection setup,
+for example, the OpenTelemetry collector [configuration][otel-collector-configuration].
+1. Upgrading and downgrading agents and management of agent-specific packages,
+including the base agent functionality and plugins.
```
Comment (cartermp) [inline content/en/docs/collector/management.md:24]:
```suggestion
1. Upgrading/downgrading agents and management of agent-specific packages,
including the base agent functionality and plugins.
```

"the base agent functionality" --> is that mean to be a part of "agent-specific packages"? If so, what's it referring to in particular? Maybe some examples would be helpful. Although IIRC you can't just manage packages in the collector, but need to build a version of the collector with components baked in.

File: content/en/docs/collector/management.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+changes in the environment or due to policy changes.
+1. Health and performance monitoring of the agents, typically CPU and memory usage
+and also agent-specific metrics, for example, the rate of processing or
+backpressure-related information.
+1. Connection management between a control plane and the agent such as
+handling of TLS certificates (revocation and rotation).
+
+Not every use case requires support for all of the above agent management tasks.
+In the context of OpenTelemetry we aim for a dogfooding, so task 
+_4. Health and performance monitoring_ is ideally done using OpenTelemetry.
```
Comment (cartermp) [inline content/en/docs/collector/management.md:35]: Hmm. I don't understand this sentence. Dogfooding in my experience is using the software an org (we) produce as a means to better empathize with end users and fix problems that we all experience. How is that relevant here?

File: content/en/docs/collector/management.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+## OpAMP
+Observability vendors and cloud providers offer proprietary solutions for
```
Comment (cartermp) [inline content/en/docs/collector/management.md:38]:
```suggestion

## OpAMP
Observability vendors and cloud providers offer proprietary solutions for 
```

File: content/en/docs/collector/management.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+## OpAMP
+Observability vendors and cloud providers offer proprietary solutions for 
+agent management. In the open source observability space, there is an emerging standard 
+that you can use for agent management: Open Agent Management Protocol (OpAMP).
+
+The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a 
+fleet of telemetry data agents. These agents can be  [OpenTelemetry collectors][otel-collector] 
+but you could also be using it for managing other agents, such as Fluent Bit or
+Prometheus in agent mode or indeed a combination of different agents, forming
+a fleet.
```
Comment (cartermp) [inline content/en/docs/collector/management.md:46]:
```suggestion
The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a 
fleet of telemetry data agents. These agents can be  [OpenTelemetry collectors][otel-collector],
Prometheus in agent mode, Fluent Bit, or other agents in any arbitrary combination.
```

File: content/en/docs/collector/management.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+* The **OpAMP server** is part of the control plane and acts as the orchestrator,
+managing a fleet of telemetry agents.
+* The **OpAMP client** us part of the data plane. The client side of OpAMP can
+be implemented in-process, for example, as the case for [OpAMP support in the OpenTelemetry
+collector][opamp-in-otel-collector]. The client side of OpAMP could alternatively
+be implemented out-of-process. For this latter option, you can use a supervisor 
+that takes care of the OpAMP specific communication with the OpAMP server and at 
+the same time controls the telemetry agent, for example to apply a configuration
+or to upgrade it. Note that the supervisor/telemetry communication
+is not part of OpAMP.
```
Comment (cartermp) [inline content/en/docs/collector/management.md:60]:
```suggestion
* The **OpAMP client** is part of the data plane. The client side of OpAMP can
be implemented in-process, for example, as the case for [OpAMP support in the OpenTelemetry
collector][opamp-in-otel-collector]. The client side of OpAMP could alternatively
be implemented out-of-process. For this latter option, you can use a supervisor 
that takes care of the OpAMP specific communication with the OpAMP server and at 
the same time controls the telemetry agent, for example to apply a configuration
or to upgrade it. Note that the supervisor/telemetry communication
is not part of OpAMP.
```

**Review by cartermp — CHANGES_REQUESTED** [review]
Some minor suggestions.

These docs feel a little weird to me right now -- the second section in particular feels like high-level conceptual docs for _implementers_ of OpAmp, but the primary readers of the docs are end-users. Should that be on the website, or in the OpAmp GitHub repo?

File: content/en/docs/collector/management.md (hunk 5 above)
Comment (mhausenblas) [reply content/en/docs/collector/management.md:35]: Maybe dogfooding is the wrong term. It's a recommendation to use and open standard (OpenTelemetry? :) for this task. Can drop if you think it's not appropriate here.

**Comment by mhausenblas** [comment]
Thanks for your feedback @cartermp and I'm happy to extend the second section with a usage example based on the Go lib, yes.

**Comment by austinlparker** [comment]
In order to end-run the entire 'agent' vs. 'auto-instrumentation' vs. 'whatever' discussion, would it be possible to just include a note that 'agent' is used as a catch-all term for opentelemetry components that respond to OpAMP, or something like it?

**Review by chalin — COMMENTED** [review]
Please run Prettier over the file: `npx prettier --ignore-path '' -w <path-to-your-file>`
Thanks!

File: content/en/docs/collector/management.md (hunk 3 above)
Comment (tigrannajaryan) [inline content/en/docs/collector/management.md:17]:
```suggestion
to manage agents. Typical agent management tasks include:
```
I think people may have different opinions about how common the particular task is, so perhaps omit the part in parenthesis.

**Review by tigrannajaryan — APPROVED** [review]

File: content/en/docs/collector/management.md (hunk 5 above)
Comment (svrnm) [reply content/en/docs/collector/management.md:35]: "OpenTelemetry on OpenTelemetry" or something like that?

File: content/en/docs/collector/management.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+ideally done using OpenTelemetry.
+
+## OpAMP
+
+Observability vendors and cloud providers offer proprietary solutions for agent
+management. In the open source observability space, there is an emerging
+standard that you can use for agent management: Open Agent Management Protocol
+(OpAMP).
+
+The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a
```
Comment (mviitane) [inline content/en/docs/collector/management.md:47]: You could consider removing the status from this page. It may become outdated when the status changes. The status is anyway visible on the spec page.

**Review by mviitane — APPROVED** [review]
LGTM, this is a very helpful page. Added only one minor comment.

File: content/en/docs/collector/management.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
+---
+title: Management
+weight: 23
+---
+
+Please be sure to review the following documentation as prerequisite:
```
Comment (chalin) [inline content/en/docs/collector/management.md:6]:
```suggestion
This document describes how you can manage your OpenTelemetry collector
deployment at scale.

To get the most out of this page you should know how to install and configure
the collector. These topics are covered elsewhere:
```

File: content/en/docs/collector/management.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+- [Getting Started][otel-collector-getting-started] to understand how to install
+  the OpenTelemetry collector.
+- [Configuration][otel-collector-configuration] for how to configure the
+  OpenTelemetry collector, setting up telemetry pipelines.
+
+This document describes how you can manage your OpenTelemetry collector
+deployment at scale.
+
```
Comment (chalin) [inline content/en/docs/collector/management.md:15]:
```suggestion
```

File: content/en/docs/collector/management.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  OpenTelemetry collector][opamp-in-otel-collector]. The client side of OpAMP
+  could alternatively be implemented out-of-process. For this latter option, you
+  can use a supervisor that takes care of the OpAMP specific communication with
+  the OpAMP server and at the same time controls the telemetry agent, for
+  example to apply a configuration or to upgrade it. Note that the
+  supervisor/telemetry communication is not part of OpAMP.
+
+Let's have a look at a concrete deployment:
+
+![OpAMP example setup](/img/docs/OTel_OpAMP.svg)
```
Comment (chalin) [inline content/en/docs/collector/management.md:72]: I know that you've moved the image already, but I'd like to ask that you relocate the image closer to where it is being used, and rename the file to follow the website's file naming conventions. So move and rename:

- `/img/docs/OTel_OpAMP.svg`

to 

- `/content/en/docs/collector/img/opamp.svg`

Then update the reference in the file.
```suggestion
![OpAMP example setup](../img/opamp.svg)
```

**Review by chalin — CHANGES_REQUESTED** [review]
Great contribution!

See inline comments for suggestions: an added page description, slightly reworked start-of-page, and image file move.

**Comment by chalin** [comment]
Also, if you could rebase, that would be grand!

File: content/en/docs/collector/management.md (hunk 10 above)
Comment (chalin) [reply content/en/docs/collector/management.md:7]: Hey @mhausenblas. This had been marked as resolved, but the changes were not applied AFAICT.

File: content/en/docs/collector/management.md (hunk 11 above)
Comment (chalin) [reply content/en/docs/collector/management.md:16]: Hey @mhausenblas. This had been marked as resolved, but the changes were not applied AFAICT. (The sentence was moved to the start of the page.

File: content/en/docs/collector/management.md (hunk 10 above)
Comment (chalin) [reply content/en/docs/collector/management.md:7]: @mhausenblas - btw, in case you didn't know: you can use the "commit" buttons that are a part of the suggestions to accept changes. It makes is easy, and less error prone, to apply changes.

**Review by cartermp — APPROVED** [review]

File: content/en/docs/collector/management.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+```
+
+Next, we need an OpenTelemetry collector binary that the OpAMP supervisor can
+manage. For that, install the [OpenTelemetry Collector Contrib][otelcolcontrib]
+distro. The path to the collector binary (where you installed it into) is
+referred to as `$OTEL_COLLECTOR_BINARY` in the following.
+
+In the `./opamp-go/internal/examples/server` directory, launch the OpAMP server:
+
+```
```
Comment (chalin) [inline content/en/docs/collector/management.md:105]:
```suggestion
```console
```

File: content/en/docs/collector/management.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+have Go in version 1.19 or above available.
+
+We will set up a simple OpAMP control plane consisting of an example OpAMP
+server and let an OpenTelemetry collector connect to it via an example OpAMP
+supervisor.
+
+First, clone the `open-telemetry/opamp-go` repo:
+
+```sh
+$ git clone https://github.com/open-telemetry/opamp-go.git
```
Comment (chalin) [inline content/en/docs/collector/management.md:95]: When there's no command output, and the code-block language is `sh` (or `shell` or `bash`), drop the prompt:

```suggestion
git clone https://github.com/open-telemetry/opamp-go.git
```

File: content/en/docs/collector/management.md [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+  pipelines:
+    metrics:
+      receivers: [hostmetrics, prometheus/own_metrics]
+      exporters: [logging]
+```
+
+Now it's time to launch the supervisor (which in turn will launch your
+OpenTelemetry collector):
+
+```
```
Comment (chalin) [inline content/en/docs/collector/management.md:162]:
```suggestion
```console
```

File: content/en/docs/collector/management.md [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+If everything worked out you should now be able to go to
+[http://localhost:4321/](http://localhost:4321/) and access the OpAMP server UI
+where you should see your collector listed, managed by the supervisor:
+
+![OpAMP example setup](../img/opamp-server-ui.png)
+
+You can also query the collector for the metrics exported (note the label
+values):
+
+```
```
Comment (chalin) [inline content/en/docs/collector/management.md:197]:
```suggestion
```console
```

**Review by chalin — APPROVED** [review]
See inline comments for code-block language fixes.
Other than, that LGTM

**Comment by mhausenblas** [comment]
@chalin thanks a bunch for the guidance (and that will be useful for future contribs as well, so really appreciate it!). I've addressed all of the feedback now, I think we're good to merge.
